src/xenia/gpu/command_processor.cc

```diff
@@ -1218,6 +1220,12 @@ bool CommandProcessor::ExecutePacketType3_DRAW_INDX(RingBuffer* reader,
     } break;
   }
 
+  auto viz_query = register_file_->Get<reg::PA_SC_VIZ_QUERY>();
+  if (viz_query.viz_query_ena && viz_query.kill_pix_post_hi_z) {
+    // TODO(Triang3l || JoelLinn) Handle this properly in the render backends.
+    return true;
```

Comment:
We can't drop draw calls this way if there's vertex shader memexport in them. Though for now that's the only edge case that needs to be handled — it's possible that depth/stencil counters may be affected somehow, though that would be a weird situation, and we don't support occlusion queries yet.